components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/util/RestClientUtil.java

```diff
@@ -0,0 +1,107 @@
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*
+*/
+package org.wso2.carbon.apimgt.usage.client.util;
+
+import org.apache.commons.codec.binary.Base64;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+
+/**
+ * Class contain the util method used by the rest client impl
+ */
+public class RestClientUtil {
+
+    private final static SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
+
+    /**
+     * conversion from date to long value
+     *
+     * @param date date in the yyyy-MM-dd format
+     * @return long value of the date
+     * @throws ParseException throw when error in parsing date
+     */
+    public static long dateToLong(String date) throws ParseException {
+        Date fDate = dateFormat.parse(date);
+        Long lDate = fDate.getTime();
+        return lDate;
+    }
+
+    /**
+     * return the ceiling time of the time and convert to long
+     *
+     * @param date date in the yyyy-MM-dd format
+     * @return long ceiling value of date
+     * @throws ParseException throw when error in parsing date
+     */
+    public static long getCeilingDateAsLong(String date) throws ParseException {
+        Date fDate = dateFormat.parse(date);
+        Calendar calender = Calendar.getInstance();
+        calender.setTime(fDate);
+        calender.set(Calendar.HOUR, 0);
+        calender.set(Calendar.MINUTE, 0);
+        calender.set(Calendar.SECOND, 0);
+        calender.set(Calendar.DATE, calender.get(Calendar.DATE) + 1);
+        return calender.getTimeInMillis();
+    }
+
+    /**
+     * return the floor time of the time and convert to long
+     *
+     * @param date date in the yyyy-MM-dd format
+     * @return long floor value of date
+     * @throws ParseException throw when error in parsing date
+     */
+    public static long getFloorDateAsLong(String date) throws ParseException {
+        Date fDate = dateFormat.parse(date);
+        Calendar calender = Calendar.getInstance();
+        calender.setTime(fDate);
+        calender.set(Calendar.HOUR, 0);
+        calender.set(Calendar.MINUTE, 0);
+        calender.set(Calendar.SECOND, 0);
+        calender.set(Calendar.DATE, calender.get(Calendar.DATE));
+        return calender.getTimeInMillis();
+    }
+
+    /**
+     * get the base 64 encoded username and password
+     *
+     * @param user username
+     * @param pass password
+     * @return encoded basic auth, as string
+     */
+    public static String encodeCredentials(String user, String pass) {
+        String cred = user + ':' + pass;
+        byte[] encodedBytes = Base64.encodeBase64(cred.getBytes());
+        return new String(encodedBytes);
+    }
+
+    /**
+     * Parsing long value to the Date format
+     * @param time time in long
+     * @return
+     */
+    public static String longToDate(long time){
```

Comment:
fixed in commit: https://github.com/wso2/carbon-apimgt/commit/9619da2271b49f37ef93456b6f8a8ac235f685ea